sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -195,7 +195,7 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry) extends
       throw new AnalysisException("Invalid number of arguments for function " + name +
         ". Expected: 0; Found: " + e.length)
     }
-    functionRegistry.createOrReplaceTempFunction(name, builder)
+    functionRegistry.createOrReplaceTempFunction(name, builder, "scala_udf")
```

Comment:
above this method, there is a big chunk of comments showing how these methods are generated. Let's update them as well.